include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -4064,122 +3667,151 @@ namespace internal
+                    continue;
+
+                  if (eval.get_all_face_orientations()[v] != 0)
+                    adjust_for_face_orientation_per_lane(
+                      dim,
+                      n_components,
+                      v,
+                      evaluation_flag,
+                      &eval.get_shape_info().face_orientations_quad(
+                        eval.get_all_face_orientations()[v], 0),
+                      false,
+                      Utilities::pow(n_q_points_1d, dim - 1),
+                      &temp[0][0],
+                      eval.begin_values(),
+                      eval.begin_gradients(),
+                      eval.begin_hessians());
+                }
+            }
+          else if (eval.get_face_orientation() != 0)
+            for (unsigned int c = 0; c < n_components; ++c)
+              adjust_for_face_orientation(
+                dim,
+                n_components,
+                evaluation_flag,
+                &eval.get_shape_info().face_orientations_quad(
+                  eval.get_face_orientation(), 0),
+                false,
+                Utilities::pow(n_q_points_1d, dim - 1),
+                temp,
+                eval.begin_values(),
+                eval.begin_gradients(),
+                eval.begin_hessians());
+        }
 
-      (void)sm_ptr;
-
-      Processor<fe_degree, n_q_points_1d> p(n_components,
-                                            false,
-                                            src_ptr,
-                                            sm_ptr,
-                                            data,
-                                            dof_info,
-                                            values_quad,
-                                            gradients_quad,
-                                            hessians_quad,
-                                            scratch_data,
-                                            evaluate_values,
-                                            evaluate_gradients,
-                                            evaluate_hessians,
-                                            active_fe_index,
-                                            first_selected_component,
-                                            cells,
-                                            face_nos,
-                                            subface_index,
-                                            dof_access_index,
-                                            face_orientations,
-                                            orientation_map);
-
-      if (n_face_orientations == VectorizedArrayType::size())
-        return fe_face_evaluation_process_and_io<VectorizedArrayType::size()>(
-          p);
+      return false;
+    }
+
+    static bool
+    supports(
+      const EvaluationFlags::EvaluationFlags evaluation_flag,
+      const MatrixFreeFunctions::ShapeInfo<VectorizedArrayType> &shape_info,
```

Comment:
sometimes I find the name `data`, sometimes `shape_info`. Why not unifying this?